File: backend/go/controller/sample_controller.go
package controller

import (
	"fmt"
	"net/http"

	"github.com/gin-gonic/gin"
)

func SetupSampleController(engine *gin.Engine) {
	routerGroup := engine.Group("/api/v0/sample")

	routerGroup.GET("/say-hi", sayHi)
	routerGroup.GET("/say-my-name", sayMyName)
}

func sayHi(context *gin.Context) {
	context.JSON(http.StatusOK, "hello world")
}

func sayMyName(context *gin.Context) {
	parameterName := context.Query("name")
	if len(parameterName) == 0 {
		context.JSON(http.StatusBadRequest, "You must pass a name parameter")
		return
	}
	context.JSON(http.StatusOK, fmt.Sprintf("hello %v!", parameterName))
}

File: backend/python/controller/sample_controller.py

from flask import Blueprint, Response, request


blueprint = Blueprint(
    'sample_controller',
    __name__,
    url_prefix='/api/v0/sample'
)

@blueprint.route('/say-hi', methods=['GET'])
def say_hi():
    return Response('hello world', status=200, mimetype='application/json')

@blueprint.route('/say-my-name', methods=['GET'])
def say_my_name():
    name = request.args.get('name')
    if len(name) == 0:
        return Response('You must pass a name parameter', status=400, mimetype='application/json')
    return Response(f'hello {name}!', status=200, mimetype='application/json')